File: Selenium/Activities/Activity16.py
import win32clipboard
from selenium import webdriver
from selenium.webdriver import ActionChains, Keys
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service) as driver:
    wait = WebDriverWait(driver, 10)
    driver.get("https://www.training-support.net/selenium/dynamic-attributes")

    print("Title of Dynamic Attributes page: ", driver.title)

    username = driver.find_element(By.XPATH, "//input[contains(@class, '-username')]")
    password = driver.find_element(By.XPATH, "//input[contains(@class, '-password')]")
    confirm_password = driver.find_element(By.XPATH,
                "//label[contains(text(), 'Confirm Password')]//following::input")
    email = driver.find_element(By.XPATH, "//input[contains(@class, 'email-')]")

    username.send_keys("First")
    password.send_keys("Password")
    confirm_password.send_keys("Password")
    email.send_keys("first_user@gmail.com")

    driver.find_element(By.XPATH, "//button[text()='Sign Up']").click()

    wait.until(expected_conditions
               .visibility_of_element_located((By.ID, "action-confirmation")))

    message = driver.find_element(By.ID, "action-confirmation").text
    print("SignUp message: ", message)


driver.close()

File: Selenium/Activities/Activity16.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.awt.datatransfer.UnsupportedFlavorException;
import java.io.IOException;
import java.time.Duration;

public class Activity16 {
    public static void main(String[] args) throws InterruptedException, IOException, UnsupportedFlavorException {
        WebDriverManager.firefoxdriver().setup();

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://www.training-support.net/selenium/dynamic-attributes");
        System.out.println("Title of Dynamic Attributes page: " + driver.getTitle());

        WebElement userName = driver.findElement(By
                .xpath("//input[contains(@class, '-username')]"));
        WebElement password = driver.findElement(By
                .xpath("//input[contains(@class, '-password')]"));
        WebElement confirmPassword = driver.findElement(By
                .xpath("//label[text() = 'Confirm Password']/following-sibling::input"));
        WebElement email = driver.findElement(By
                .xpath("//label[contains(text(), 'mail')]/following-sibling::input"));

        userName.sendKeys("DemoUser");
        password.sendKeys("Password");
        confirmPassword.sendKeys("Password");
        email.sendKeys("demo_email@xyz.com");

        driver.findElement(By.xpath("//button[contains(text(), " +
                "'Sign Up')]")).click();

        String signUpMessage = driver.findElement(By.id("action-confirmation"))
                .getText();
        System.out.println("SignUp message: " + signUpMessage);

        driver.close();




    }
}
